pkg/util/util.go

```diff
@@ -221,16 +221,23 @@ func Md5sum(filePath string) (string, error) {
 
 // GetUsableSpace calculates usable space to use taking file system overhead into account
 func GetUsableSpace(filesystemOverhead float64, availableSpace int64) int64 {
-	// +1 always rounds up.
-	spaceWithOverhead := int64(math.Ceil((1 - filesystemOverhead) * float64(availableSpace)))
+	// Reverse the overhead calculation
+	spaceWithoutOverhead := int64(math.Ceil(float64(availableSpace) / (1 + filesystemOverhead)))
 	// qemu-img will round up, making us use more than the usable space.
 	// This later conflicts with image size validation.
-	return RoundDown(spaceWithOverhead, DefaultAlignBlockSize)
+	return RoundDown(spaceWithoutOverhead, DefaultAlignBlockSize)
 }
 
-func CalculateOverheadSpace(filesystemOverhead float64, availableSpace int64) int64 {
-	spaceWithOverhead := int64(math.Ceil(float64(availableSpace) / (1 - filesystemOverhead)))
-	return RoundUp(spaceWithOverhead, DefaultAlignBlockSize)
+// GetRequiredSpace calculates space required taking file system overhead into account
+func GetRequiredSpace(filesystemOverhead float64, requestedSpace int64) int64 {
+	// the `image` has to be aligned correctly, so the space requested has to be aligned to
+	// next value that is a multiple of a block size
+	alignedSize := RoundUp(requestedSpace, DefaultAlignBlockSize)
```

Comment:
shouldn't the alignment happen after the calculation below? Don't think we can guarantee alignment after the calculation

Reply:
I was torn between keeping this behavior (alignment before inflation) or following the one used in `CalculateOverheadSpace` (alignment after inflation). I ended up choosing this because we're aligning the size of the virtual image itself (which is what we care about) and then inflating it to account for the overhead, which I assume doesn't need to be aligned.

Reply:
Say you have a 1G disk, so it is already aligned. If fs overhead is 0.03, the output of the calculation will not be aligned

Reply:
okay